spynl/main/exceptions.py

```diff
@@ -56,12 +61,56 @@ def make_response(self):
         return response
 
     def __str__(self):
-        """This will return a str version of the message. If the message is a
+        """
+        This will return a str version of the message. If the message is a
         SpynlTranslationString, it will return an interpolated version of the
         default (no translation).
         """
         return str(self.message)
 
+    # Dictonary of exception mappings, keys are the __names__ of external
+    # exception classes, values are the corresponding internal exception classes
+    _exception_mapping = {}
+
+    @classmethod
+    def this_exception_maps(cls, external_class):
+        """
+        A decorator to map the decorated SpynlException class to an external
+        exception class.
+        """
+        def decorator(internal_class):
+            cls._exception_mapping[external_class.__name__] = internal_class
+            return internal_class
+        return decorator
+
+    @classmethod
+    def get_mapped_exception(cls, external_exception):
```

Comment:
In dislike this_exception_wraps. Not a verb and it is not a semantic counterpart to get_mapped_exception.